catalog/task/push-sbom-to-pyxis/0.1/push-sbom-to-pyxis.yaml

```diff
@@ -12,24 +12,89 @@ spec:
   description: >-
     Tekton task that extracts sbom from a pull spec and pushes it to Pyxis.
   params:
-    - name: imageURL
-      description: URL of the main binary image.
+    - name: imageURLs
+      description: Space separated list of binary image URLs that are used to obtain SBOMs via cosign
       type: string
+    - name: containerImageIDs
+      description: Space separated list of Pyxis image IDs
+      type: string
+    - name: pyxisSecret
+      type: string
+      description: "The kubernetes secret to use to authenticate to Pyxis. It needs to contain two keys: key and cert."
+    - name: server
+      type: string
+      description: The server type to use. Options are 'production' and 'stage'
+      default: production
   volumes:
     - name: workdir
       emptyDir: {}
   steps:
-    - name: download-sbom-for-image
-      # Image taken from https://github.com/sigstore/cosign
-      # and copied to Quay
+    - name: download-sbom-files
+      image:
+        quay.io/hacbs-release/release-utils:62f3c9ca85e45a4a76f2fb5c0ec9402f3cc08b2d
+      volumeMounts:
+        - mountPath: /workdir
+          name: workdir
+      script: |
+        #!/usr/bin/env sh
+        set -eux
+
+        IMAGEURLS=($(params.imageURLs))
+        IMAGEIDS=($(params.containerImageIDs))
+
+        if [ ${#IMAGEURLS[@]} != ${#IMAGEIDS[@]} ]; then
+          echo "ERROR: Lengths of imageURLs and containerImageIDs do not match"
+          exit 1
+        fi
+
+        mkdir /workdir/sboms
+        cd /workdir/sboms
+
+        for i in ${!IMAGEURLS[@]}; do
+          echo "Fetching sbom for image: ${IMAGEURLS[$i]}"
+          cosign download sbom --output-file ${IMAGEIDS[$i]}.json ${IMAGEURLS[$i]}
+        done
+
+    - name: push-sbom-files-to-pyxis
       image:
-        quay.io/redhat-appstudio/cosign:v1.13.1
-      args:
-        - download
-        - sbom
-        - --output-file
-        - /workdir/sbom-cyclonedx.json
-        - "$(params.imageURL)"
+        quay.io/hacbs-release/release-utils:62f3c9ca85e45a4a76f2fb5c0ec9402f3cc08b2d
+      env:
+        - name: pyxisCert
+          valueFrom:
+            secretKeyRef:
+              name: $(params.pyxisSecret)
+              key: cert
+        - name: pyxisKey
+          valueFrom:
+            secretKeyRef:
+              name: $(params.pyxisSecret)
+              key: key
       volumeMounts:
         - mountPath: /workdir
           name: workdir
+      script: |
+        #!/usr/bin/env sh
+        set -eu
+
+        if [[ "$(params.server)" == "production" ]]
+        then
+          PYXIS_GRAPHQL_API="https://graphql-pyxis.api.redhat.com/"
+        elif [[ "$(params.server)" == "stage" ]]
+        then
+          PYXIS_GRAPHQL_API="https://graphql-pyxis.preprod.api.redhat.com/"
```

Comment:
Instead of using $param.server point to stage or prod, would not be better have the URL itself as a parameter?

Reply:
Maybe, but I'm using the same approach as `catalog/task/create-pyxis-image/0.3/create-pyxis-image.yaml` here - I think it's good to stay consistent.

Reply:
I understand the point although I think that in case (for any reason) the URL changes, or in case of a big downtime in pyxis that would require a change to a temp url, , we would need to release a new version of the script. I would prefer just change the parameter in the strategy.

Reply:
That's a good point. But on the other hand I don't think it's likely that urls such as Pyxis endpoints will change often - Pyxis is used by many things and it cannot just do that without some big announcements and time for users to adapt etc. So I don't think it's a big concern.

@davidmogar wdyt of this?

Reply:
I am ok going this way if all agree. For the record, I am a pessimist - or maybe my former sysadmin career is talking louder - that what is unlikely to happen, will happen sooner or later. The most important is being aware. If it happens we at least already know how to fix it.